pkg/cmd/server/admin/create_nodeconfig.go

```diff
@@ -0,0 +1,375 @@
+}
+
+func (o CreateNodeConfigOptions) MakeClientCert(clientCertFile, clientKeyFile string) error {
+	if o.IsCreateClientCertificate() {
+		createNodeClientCert := CreateNodeClientCertOptions{
+			GetSignerCertOptions: o.GetSignerCertOptions,
+			CertFile:             clientCertFile,
+			KeyFile:              clientKeyFile,
+			NodeName:             o.NodeName,
+		}
+		if err := createNodeClientCert.Validate(nil); err != nil {
+			return err
+		}
+		if _, err := createNodeClientCert.CreateNodeClientCert(); err != nil {
+			return err
+		}
+
+	} else {
+		if err := CopyFile(o.ClientCertFile, clientCertFile, 0644); err != nil {
+			return err
+		}
+		if err := CopyFile(o.ClientKeyFile, clientKeyFile, 0600); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeServerCert(serverCertFile, serverKeyFile string) error {
+	if o.IsCreateServerCertificate() {
+		nodeServerCertOptions := CreateServerCertOptions{
+			GetSignerCertOptions: o.GetSignerCertOptions,
+
+			CertFile: serverCertFile,
+			KeyFile:  serverKeyFile,
+
+			Hostnames: o.Hostnames,
+		}
+
+		if err := nodeServerCertOptions.Validate(nil); err != nil {
+			return err
+		}
+		if _, err := nodeServerCertOptions.CreateServerCert(); err != nil {
+			return err
+		}
+
+	} else {
+		if err := CopyFile(o.ServerCertFile, serverCertFile, 0644); err != nil {
+			return err
+		}
+		if err := CopyFile(o.ServerKeyFile, serverKeyFile, 0600); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeCA(clientCopyOfCAFile string) error {
+	if err := CopyFile(o.APIServerCAFile, clientCopyOfCAFile, 0644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeKubeConfig(clientCertFile, clientKeyFile, clientCopyOfCAFile, kubeConfigFile string) error {
+	createKubeConfigOptions := CreateKubeConfigOptions{
+		APIServerURL:    o.APIServerURL,
+		APIServerCAFile: clientCopyOfCAFile,
+		ServerNick:      "master",
+
+		CertFile: clientCertFile,
+		KeyFile:  clientKeyFile,
+		UserNick: "node",
+
+		KubeConfigFile: kubeConfigFile,
+	}
+	if err := createKubeConfigOptions.Validate(nil); err != nil {
+		return err
+	}
+	if _, err := createKubeConfigOptions.CreateKubeConfig(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeNodeConfig(serverCertFile, serverKeyFile, kubeConfigFile, nodeConfigFile string) error {
+	config := &configapi.NodeConfig{
+		NodeName: o.NodeName,
+
+		ServingInfo: configapi.ServingInfo{
+			BindAddress: net.JoinHostPort(o.ListenAddr.Host, strconv.Itoa(ports.KubeletPort)),
+		},
+
+		VolumeDirectory:       o.VolumeDir,
+		NetworkContainerImage: o.NetworkContainerImage,
+		AllowDisabledDocker:   o.AllowDisabledDocker,
+
+		DNSDomain: o.DNSDomain,
+		DNSIP:     o.DNSIP,
+
+		MasterKubeConfig: kubeConfigFile,
+	}
+
+	if o.UseTLS() {
+		config.ServingInfo.ServerCert = configapi.CertInfo{
+			CertFile: serverCertFile,
+			KeyFile:  serverKeyFile,
+		}
```

Comment:
> I can add that in my node https PR if you want

Yeah, let's add that separately.  It'll need to be another parameter since it doesn't necessarily match the server's CA.